pipx/pipxrc.py

```diff
@@ -0,0 +1,196 @@
+        self.package_or_url: Optional[str] = None
+        self.install: InstallOptions = InstallOptions(
+            pip_args=None, venv_args=None, include_dependencies=None
+        )
+        self.venv_metadata: Optional[PipxVenvMetadata] = None
+        self.injected_packages: Optional[Dict[str, InjectedPackage]] = None
+        self._pipxrc_version: str = "0.1"
+
+    def to_dict(self) -> Dict[str, Any]:
+        venv_metadata: Optional[Dict[str, Any]]
+        injected_packages: Optional[Dict[str, Dict[str, Any]]]
+
+        if self.venv_metadata is not None:
+            venv_metadata = self.venv_metadata._asdict()
+        else:
+            venv_metadata = None
+        if self.injected_packages is not None:
+            injected_packages = {
+                k: v._asdict() for (k, v) in self.injected_packages.items()
+            }
+        else:
+            injected_packages = None
+
+        return {
+            "package_or_url": self.package_or_url,
+            "install": self.install._asdict(),
+            "venv_metadata": venv_metadata,
+            "injected_packages": injected_packages,
+            "pipxrc_version": self._pipxrc_version,
+        }
+
+    def from_dict(self, pipxrc_info_dict) -> None:
+        self.package_or_url = pipxrc_info_dict["package_or_url"]
+        self.install = InstallOptions(**pipxrc_info_dict["install"])
+        self.venv_metadata = PipxVenvMetadata(**pipxrc_info_dict["venv_metadata"])
+        self.injected_packages = {
+            k: InjectedPackage(**v)
+            for (k, v) in pipxrc_info_dict["injected_packages"].items()
+        }
+
+
+class Pipxrc:
+    def __init__(self, venv_dir: Path, read: bool = True):
+        self.venv_dir = venv_dir
+        self.pipxrc_info = PipxrcInfo()
+        if read:
+            self.read()
+
+    def reset(self) -> None:
+        self.pipxrc_info = PipxrcInfo()
+
+    def _val_or_default(self, value: Optional[Multi], default: Multi) -> Multi:
+        if value is not None:
+            return value
+        else:
+            return default
+
+    def get_package_or_url(self, default: str) -> str:
+        return self._val_or_default(self.pipxrc_info.package_or_url, default)
+
+    def get_install_pip_args(self, default: List[str]) -> List[str]:
+        return self._val_or_default(self.pipxrc_info.install.pip_args, default)
+
+    def get_install_venv_args(self, default: List[str]) -> List[str]:
+        return self._val_or_default(self.pipxrc_info.install.venv_args, default)
+
+    def get_install_include_dependencies(self, default: bool) -> bool:
+        return self._val_or_default(
+            self.pipxrc_info.install.include_dependencies, default
+        )
+
+    def get_venv_metadata(self, default: PipxVenvMetadata) -> PipxVenvMetadata:
+        return self._val_or_default(self.pipxrc_info.venv_metadata, default)
+
+    def get_injected_packages(
+        self, default: List[Dict[str, Any]]
+    ) -> List[Dict[str, Any]]:
```

Comment:
Here it is a list of Dicts, and in another place it's just a Dict. Can it always just be a Dict?

Reply:
True, this is weird.  Let me look into it.

Reply:
Done.  Now it's always a dict of `InjectedPackage`s